redistimeseries/client.py

```diff
@@ -314,10 +314,11 @@ def pipeline(self, transaction=True, shard_hint=None):
         Overridden in order to provide the right client through the pipeline.
         """
         p = Pipeline(
-            connection_pool=self.connection_pool,
-            response_callbacks=self.response_callbacks,
+            connection_pool=self.redis.connection_pool,
+            response_callbacks=self.redis.response_callbacks,
             transaction=transaction,
             shard_hint=shard_hint)
+        p.redis = p
```

Comment:
@ashtul why do you need `        p.redis = p`?

Reply:
since we use `self.redis.execute_command(...)`
similar to `self.redis = conn if conn is not None else Redis(*args, **kwargs)`
We keep the connection as part of the object which doesn't inherit from `Redis` anymore.

Reply:
:+1: